test/Ndb/NdbSpec.hs

```diff
@@ -0,0 +1,76 @@
+module Ndb.NdbSpec (spec) where
+
+import Control.Carrier.Diagnostics (runDiagnostics)
+import Control.Carrier.Stack (runStack)
+import Effect.Exec (runExecIO)
+import Effect.ReadFS (runReadFSIO)
+import Path (Abs, File, Path)
+import Path.IO qualified as PIO
+import ResultUtil (assertOnSuccess)
+import Strategy.NDB.Internal (NdbEntry (..), readNDB)
+import Test.Hspec (Spec, describe, it, runIO, shouldBe)
+
+spec :: Spec
+spec = do
+  describe "Container NDB parser" $ do
+    target <- runIO testDbFile
+    result <- runIO . runStack . runDiagnostics . runExecIO . runReadFSIO $ readNDB target
+
+    it "parses ndb contents" $
+      assertOnSuccess result $ \_ c ->
+        c `shouldBe` expectedEntries
+
+-- | Test file from: https://github.com/jssblck/go-rpmdb/blob/9a3bd2ebb923399db2189c5d8963af27368ba2c8/pkg/rpmdb_test.go#L72-L74
```

Comment:
The personal fork is the code I was actually referencing, and is the one that contains changes I made to generate testing blobs, so I wanted to be maximally correct.

... but I should (and will) swap them to the root, since those modifications mattered for the BDB parser but not for this parser. Thanks for pointing that out.